apis/core-api/src/core/controllers/RoomCommentController/RoomCommentController.ts

```diff
@@ -0,0 +1,125 @@
+import { CommentTypeEnum, OrderEnum } from '@ultras/utils';
+import BaseController from 'core/controllers/BaseController';
+import { RoomService, CommentService } from 'core/services';
+import { ResourceNotFoundError } from 'modules/exceptions';
+
+import { DEFAULT_PAGINATION_ATTRIBUTES } from '@constants';
+import type {
+  CommentsListParams,
+  CommentListResult,
+  CommentCreateParamsInterface,
+  CommentUpdateParamsInterface,
+  CommentDeleteParamsInterface,
+} from './types';
+
+class RoomCommentController extends BaseController {
+  /**
+   * Add comment under room.
+   */
+  static async create({ userId, roomId, content }: CommentCreateParamsInterface) {
+    // get room model
+    const room = await RoomService.getById({ id: roomId }, false);
+    if (!room) {
+      throw new ResourceNotFoundError({
+        message: 'Room not found.',
+      });
+    }
+
+    // add comment under post
+    const comment = await CommentService.comment({
+      resourceType: CommentTypeEnum.post,
+      resourceId: room.getDataValue('postId'),
+      userId,
+      content,
+    });
+
+    return comment;
+  }
+
+  /**
+   * Update comment under room.
+   */
+  static async update({
+    userId,
+    roomId,
+    commentId,
+    content,
+  }: CommentUpdateParamsInterface) {
+    // get room model
+    const room = await RoomService.getById({ id: roomId }, false);
+    if (!room) {
+      throw new ResourceNotFoundError({
+        message: 'Room not found.',
```

Comment:
you mean client side need to call to another service when he got error? I think is not a good idea to make another call to get translation, maybe we can user that service in server side?

Reply:
Lets do this just for real public sdks, to provide flexibility for public use, just we are gonna use an internationalization server, and its another bad idea to do call when raising an error from the back-end, in client side it can be only once. Just need to get all key values for the selected language, but yeah, need to think a bit for scalability, going ahead with no translations right now since initial version supports only for English